inc/scaip-shortcode-inserter.php

```diff
@@ -70,12 +70,22 @@ function( $blocks, $block ) {
 
 	$output = '';
 
+	$blocks_allowing_insertion = array_flip( apply_filters( 'scaip_allowing_insertion_blocks', [ 'core/paragraph' ] ) );
+
 	foreach ( $parsed_blocks as $block ) {
 		$is_empty = empty( trim( $block['innerHTML'] ) );
 		if ( $is_empty ) {
 			continue;
 		}
 
+		/**
+		 * Skip insertion if the block is not on the allowing-insertion list.
+		 */
+		if ( ! isset( $blocks_allowing_insertion[ $block['blockName'] ] ) ) {
```

Comment:
⚠️   Does this actually skip insertion? The `!` here makes this look to me like it explicitly inserts the SCAIP block if the present block is **not** in the list of blocks allowing insertion?

Reply:
If the `blockName` is not set on the `$blocks_allowing_insertion`, it will append the block content to the `$output` and `continue` the loop, therefore, preserve the original content without incrementing the `$inserted_shortcode_index` nor appending the Ad shortcode.